homeassistant/components/stt/__init__.py

```diff
@@ -71,6 +72,13 @@
 @callback
 def async_default_engine(hass: HomeAssistant) -> str | None:
     """Return the domain or entity id of the default engine."""
+    entity_registry = er.async_get(hass)
+    if (
+        cloud_entity_id := entity_registry.async_get_entity_id(
+            DOMAIN, "cloud", "cloud-speech-to-text"
+        )
+    ) and cloud_entity_id in hass.states.async_entity_ids(DOMAIN):
+        return cloud_entity_id
```

Comment:
I think it would be cleaner to move this to a function in `cloud/stt.py` so stt doesn't need to know about the unique id and so on.

Also, `stt.async_default_engine` will be called from here when the pipeline store is first setup: `https://github.com/home-assistant/core/blob/f57581a3f94ed9396ceff678a0585c76c64c8645/homeassistant/components/assist_pipeline/pipeline.py#L1409`

That will happen before `cloud` is setup if the user has `default_config` enabled.

Finally, we create a cloud enabled pipeline when cloud is setup: https://github.com/home-assistant/core/pull/91991